libs/entry-components/configure-theming.md

```diff
@@ -0,0 +1,435 @@
+
+```scss
+$custom-theme: (
+ general: (
+ density: 0
+ )
+);
+```
+
+#### Colors
+
+Defines various color properties:
+
+- `primary`: Main color used in application, it sets the tone for the overall theme
+- `accent`: Highlight certain elements and create visual interest
+- `font`: Specifies the default font color.
+- `disabled`: Style elements that are in a disabled state:
+  - `foreground`: Text color for disabled elements
+  - `background`: Background color for disabled elements
+
+```scss
+$custom-theme: (
+ general: (
+  density: 0,
+  colors: (
+   primary: #2581C4,
+   accent: #EA518D,
+   font: #323232,
+   disabled: (
+    foreground: rgb(0 0 0 / .38),
+    background: rgb(0 0 0 / .12)
+   )
+  )
+ )
+);
+```
+
+#### Fonts
+
+Allow customization of typography and font styles based od Angular Material typography levels. Each font related property in configuration can have **family** and **size** values.
+
+- `hero-titles`: Define typography for h1, h2, h3, h4 elements.
+- `titles`: Typography h3, h4 elements
+- `body`: Define typography for base body text.
+- `buttons`: Typography for buttons and anchors
+
+> **_NOTE:_**  Don't forget to check if you [corrected configure](#importing-fonts) or import font family's
+
+```scss
+$custom-theme: (
+ general: (
+  density: 0,
+  colors: (
+   primary: #2581C4,
+   accent: #EA518D,
+   font: #323232,
+   disabled: (
+    foreground: rgb(0 0 0 / .38),
+    background: rgb(0 0 0 / .12)
+   )
+  ),
+  fonts: (
+   hero-titles: (
+    family: 'Helvetica',
+    size: 40px
+   ),
+   titles: (
+    family: 'Roboto',
+    size: 25px
+   ),
+   body: (
+    family: 'Helvetica',
+    size: 20px
+   ),
+   buttons: (
+    family: 'Roboto',
+    size: 10px
+   )
+  ),
+ )
+);
+```
+
+To apply it on whole body of the app, add `.mat-body` class to the body element in index.html file in app root
+
+```html
+<!-- Applying mat-body class to the root component in index.html -->
+<body class="mat-body">
+<app-root></app-root>
+</body>
+```
+
+Since we're overriding Angular Material, it is important to add CSS classes for mixin that emits styles for native header elements scoped within the .mat-typography CSS class
+
+| CSS class      | Example                                       |
+| -------------- | --------------------------------------------- |
+| .mat-headline-1   | `<h1 class="mat-headline-1">{{title}}</h1>`|
+| .mat-headline-2   | `<h2 class="mat-headline-2">{{title}}</h2>`|
+
+#### Spacing
+
+The submap providing the necessary spacing information for styling the **entry form and search form** **button** and **form field**
+
+```scss
+$custom-theme: (
+ general: (
+  spacing: (
+   default: 15px
+  ),
+ )
+);
+```
+
+#### Buttons
+
+Additional customization  for buttons:
+
+- `icon-size` - defines the default icon size for buttons.
+
+Code Example of general submap configuration:
+
+```scss
+$custom-theme: (
+ general: (
+  buttons: (
+   icon-size: 48px
+  )
+ )
+)
+```
+
```

Comment:
If I haven't missed something, we are not mentioning that any property can be skipped and if done so, default value for that property will be used (unless specified differently as for fonts i.e.)

Reply:
I've included a note for that, and in the concluding part I've added a table displaying default values.